### PR TITLE
Fixed NoLikelyBOLDBug

### PR DESCRIPTION
**Pull request by handwerkerd**
@jbteves noticed a bug. In the current tedana, if no accepted components are identified then ICA is repeated with a different seed. In this branch, since ignored components were replaced by accepted + a descriptive classification tag, this repetition would never happen. I confirmed that everything that used to be `accepted` now contains the `Likely BOLD` classification_tag

Changes:
- New LikelyBOLD_comps and n_LikelyBOLD_comps properties added to component_selector.py
- Tests for all the properties (which verify results) were added
- tedana.py will now repeat ICA if no components contain the `Likely BOLD` classification_tag
- tedana_reclassify.py will still give a warning based on no accepted components rather than Likely BOLD tags. (The logic is, if reclassification as removed everything accepted, that's the warning. Reclassification will never add or remove a `Likely BOLD` tag so it's not a useful check there.
- Documentation updated to note that `Likely BOLD` is a required classification_tag

Notes:
- I confirmed this is doing what it's supposed to, but I can't replicate the integration test where this first occurred so @jbteves should do that & make sure the results now match. (Maybe make a more consistent integration test for this issue?)
- Take a look at the LikelyBOLD_comps property. pd Dataseries have some weird edge cases in my hands to make sure I didn't mess anything up.
- One other general thing for this PR. In every file, the `component_selector` is called `selector` except in tedana.py where it's called `ica_selection` Any reason NOT to change that to `selector` for consistency?

### CODE REVIEW

**Comment by jbteves** [comment]
We could in theory also have a `pca_selector`, though in other places things are agnostic to the type of selector, as I recall. That's why I called it `ica_selector` originally.

File: tedana/selection/component_selector.py [hunk 1]
```diff
@@ -509,9 +509,27 @@ def n_comps(self):
         """The number of components in the component table."""
         return len(self.component_table)
 
+    @property
+    def LikelyBOLD_comps(self):
+        """A boolean pd.DataSeries of components that are tagged "Likely BOLD"."""
+        LikelyBOLD_comps = self.component_table["classification_tags"].copy()
+        print(LikelyBOLD_comps)
```
Comment (jbteves) [inline tedana/selection/component_selector.py:516]:
```suggestion
        print(LikelyBOLD_comps)
```

Comment (jbteves) [inline tedana/selection/component_selector.py:516]: Delete print

File: tedana/selection/component_selector.py [hunk 2]
```diff
@@ -509,9 +509,27 @@ def n_comps(self):
...
+        for idx in range(len(LikelyBOLD_comps)):
+            if "Likely BOLD" in LikelyBOLD_comps.loc[idx]:
+                LikelyBOLD_comps.loc[idx] = True
+            else:
+                LikelyBOLD_comps.loc[idx] = False
+        print(LikelyBOLD_comps)
```
Comment (jbteves) [inline tedana/selection/component_selector.py:522]: Delete print

**Comment by jbteves** [comment]
Technically the bug exists in `main`, because there is a case (which is not really supposed to happen but apparently does) where you only have "ignored" and "rejected" components, but what's checked is the number of "accepted" components.

**Review by jbteves — APPROVED** [review]
LGTM. Thanks for adding a test!
